src/os/ui/timelinepanel.js

```diff
@@ -52,6 +53,18 @@ os.ui.resizeMap = 'resizeMap';
 os.ui.TimelinePanelCtrl = function($scope, $element, $timeout) {
   os.ui.TimelinePanelCtrl.base(this, 'constructor', $scope, $element, $timeout);
 
+  /**
+   * @type {boolean}
+   */
+  this['locked'] = os.time.TimelineController.getInstance().getLock();
+
+  /**
+   * @type {?angular.Scope}
+   */
+  this['scope'] = $scope;
+
+  this.tlc.listen(os.time.TimelineEventType.LOCK_TOGGLE, goog.partial(this.setLock, true), false, this);
```

Comment:
The `goog.partial` call can be removed from this listener. That injects a param into the callback, but you're not using it. It will also prevent you from cleaning up the listener unless you assign the returned function to a param, so you can pass that instance of the function to `unlisten`. This line should be:
`this.tlc.listen(os.time.TimelineEventType.LOCK_TOGGLE, this.setLock, false, this);`

That said, you also need a corresponding `unlisten` in the `destroy` function. Please add that.